File: test/Integration/test-java/src/main/java/com/function/ProductsColumnTypesTrigger.java
/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 */

package com.function;

import com.function.Common.SqlChangeProduct;
import com.google.gson.Gson;
import com.microsoft.azure.functions.ExecutionContext;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.sql.annotation.SQLTrigger;

import java.util.logging.Level;

public class ProductsColumnTypesTrigger {
    @FunctionName("ProductsColumnTypesTrigger")
    public void run(
            @SQLTrigger(
                name = "changes",
                tableName = "[dbo].[ProductsColumnTypes]",
                connectionStringSetting = "SqlConnectionString")
                SqlChangeProduct[] changes,
            ExecutionContext context) throws Exception {

        context.getLogger().log(Level.INFO, "SQL Changes: " + new Gson().toJson(changes));
    }
}

File: test-outofproc/ProductsColumnTypesTrigger.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System.Collections.Generic;
using DotnetIsolatedTests.Common;
using Microsoft.Extensions.Logging;
using Microsoft.Azure.Functions.Worker;
using Microsoft.Azure.Functions.Worker.Extensions.Sql;

namespace DotnetIsolatedTests
{
    public static class ProductsColumnTypesTrigger
    {
        /// <summary>
        /// Simple trigger function used to verify different column types are serialized correctly.
        /// </summary>
        [Function(nameof(ProductsColumnTypesTrigger))]
        public static void Run(
            [SqlTrigger("[dbo].[ProductsColumnTypes]", "SqlConnectionString")]
            IReadOnlyList<SqlChange<ProductColumnTypes>> changes,
            FunctionContext context)
        {
            ILogger logger = context.GetLogger("ProductsColumnTypesTrigger");
            logger.LogInformation("SQL Changes: " + Utils.JsonSerializeObject(changes));
        }
    }
}